c++/src/kj/common-test.c++

```diff
@@ -596,5 +596,31 @@ KJ_TEST("kj::range()") {
   KJ_EXPECT(expected == 8);
 }
 
+KJ_TEST("kj::defer()") {
+  bool executed;
+
+  // rvalue reference
+  {
+    executed = false;
+    kj::defer([&executed]() {
+      executed = true;
+    });
+  }
```

Comment:
Maybe add `KJ_EXPECT(!executed)` before leaving the block, to prove that `defer` isn't executing synchronously.

(As written right now, it is executing synchronously since you didn't assign the defer object to a variable. :) )

Reply:
Good catch! Done.